dbms/src/Flash/Coprocessor/ExecutionSummaryCollector.cpp

```diff
@@ -89,6 +108,28 @@ void ExecutionSummaryCollector::addExecuteSummaries(tipb::SelectResponse & respo
                 current.time_processed_ns = std::max(current.time_processed_ns, p_stream->getProfileInfo().execution_time);
                 current.num_produced_rows += p_stream->getProfileInfo().rows;
                 current.num_iterations += p_stream->getProfileInfo().blocks;
+
+                if (!get_storage_info)
+                {
+                    // There may be multiple UnorderedInputStream in the same executor, but they share the same table_scan_context,
+                    // Thus we only calculate the table_scan_context once;
```

Comment:
Can we hold the dm_context in DAGContext and don't need to get from BlockInputStream?

Reply:
Yes, I change to hold scan_context in DAGContext.